Record Puma pool capacity value

schneems opened the pull request:
Related to https://github.com/puma/puma/pull/1579

apg commented on lib/barnes/instruments/puma_backlog.rb:
```diff
@@ -26,22 +26,28 @@ def json_stats
+
+      def instrument!(state, counters, gauges)
+        return if json_stats.empty?
 
```
I'm not a ruby perf expert, but would think that `stat_value_from_key` could be rewritten to avoid multiple map / reduce pairs. Not sure how much it matters though, given the number of puma metrics is probably small.

schneems commented:
I updated so that it does not report the pool capacity as "0" when in fact it is not present. I also pulled out some of the logic into a class that can be unit tested. I removed the map/reduce logic and made it increment a value instead.

I'm currently running this commit live in production with `issuetriage`. We likely need to figure out a way to get some integration style tests in the future.
